bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_09/A5_09.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    // CO2
+    protected double getUnscaledCO2Min() {
+        return 0;
+    }
+
+    protected double getUnscaledCO2Max() {
+        return 255;
+    }
+
+    protected double getScaledCO2Min() {
+        return 0;
+    }
+
+    protected double getScaledCO2Max() {
+        return 2550;
+    }
+
+    protected int getUnscaledCO2Value() {
+        return getDB_2Value();
+    }
+
+    // Temperature
+    protected double getUnscaledTemperatureMin() {
+        return 0;
+    }
+
+    protected double getUnscaledTemperatureMax() {
+        return 255;
+    }
+
+    protected double getScaledTemperatureMin() {
+        return 0;
+    }
+
+    protected double getScaledTemperatureMax() {
+        return 51;
+    }
+
+    protected int getUnscaledTemperatureValue() {
+        return getDB_1Value();
+    }
+
+    // Humidity
+    protected double getUnscaledHumidityMin() {
+        return 0;
+    }
+
+    protected double getUnscaledHumidityMax() {
+        return 200;
+    }
+
+    protected double getScaledHumidityMin() {
+        return 0;
+    }
+
+    protected double getScaledHumidityMax() {
+        return 100;
+    }
+
+    protected int getUnscaledHumidityValue() {
+        return getDB_3Value();
+    }
+
+    // Battery
+    protected double getUnscaledBatteryVoltageMin() {
+        return 0;
+    }
+
+    protected double getUnscaledBatteryVoltageMax() {
+        return 255;
+    }
+
+    protected double getScaledBatteryVoltageMin() {
+        return 0;
+    }
+
+    protected double getScaledBatteryVoltageMax() {
+        return 5.1;
+    }
+
+    private int getUnscaledBatteryVoltageValue() {
+        return getDB_3Value();
+    }
+
+    @Override
+    protected State convertToStateImpl(String channelId, String channelTypeId,
+            Function<String, State> getCurrentStateFunc, Configuration config) {
+
+        if (channelId.equals(CHANNEL_CO)) {
+            double scaledCO = EEPHelper.calculateScaledValue(getUnscaledCOValue(), getScaledCOMin(), getScaledCOMax(),
+                    getUnscaledCOMin(), getUnscaledCOMax());
+            return new QuantityType<>(scaledCO, Units.PARTS_PER_MILLION);
+        } else if (channelId.equals(CHANNEL_CO2)) {
+            double scaledCO2 = EEPHelper.calculateScaledValue(getUnscaledCO2Value(), getScaledCO2Min(),
+                    getScaledCO2Max(), getUnscaledCO2Min(), getUnscaledCO2Max());
+            return new QuantityType<>(scaledCO2, Units.PARTS_PER_MILLION);
+        } else if (channelId.equals(CHANNEL_TEMPERATURE)) {
+            double scaledTemp = EEPHelper.calculateScaledValue(getUnscaledTemperatureValue(), getScaledTemperatureMin(),
+                    getScaledTemperatureMax(), getUnscaledTemperatureMin(), getUnscaledTemperatureMax());
+            return new QuantityType<>(scaledTemp, SIUnits.CELSIUS);
+        } else if (channelId.equals(CHANNEL_HUMIDITY)) {
+            return new DecimalType((getUnscaledHumidityValue() * 100.0) / getUnscaledHumidityMax());
```

Comment:
Future enhancement proposal (not for this PR): If humidity channel would be redefined to `Number:Dimensionless` this would be `QuantityType` with `Units.PERCENT`.